### PR TITLE
DATAUP-614 (related) migrate common/lang -> util/util

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

As talked about here: [#2593 comment](https://github.com/kbase/narrative/pull/2592#discussion_r750329883) , this PR moves the `common/lang` utility module to the `util` directory with the rest of them. It updates all the other module that used it to the new resource.

ESlint and Prettier threw a collective tantrum as well, so most of the code changes are due to that. Will comment inline.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* should be no visible changes
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/buildTools/loadAppWidgets.js [hunk 1]
```diff
@@ -81,7 +81,6 @@ define([
     'common/html',
     'common/jobs',
     'common/jupyter',
-    'common/lang',
```
Comment (briehl) [inline kbase-extension/static/buildTools/loadAppWidgets.js:84]: File renaming.

File: kbase-extension/static/kbase/js/common/monoBus.js [hunk 2]
```diff
@@ -14,7 +14,7 @@
  *
  *
  */
-define(['uuid', 'bluebird', 'underscore', './lang'], (Uuid, Promise, _, lang) => {
+define(['uuid', 'bluebird', 'underscore', 'util/util'], (Uuid, Promise, _, Util) => {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/monoBus.js:17]: Most changes will look like this one - `lang` -> `Util`

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js [hunk 3]
```diff
@@ -30,18 +30,15 @@ define([
     // Constants
     const t = html.tag,
         div = t('div'),
-        span = t('span'),
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js:33]: Unused.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js [hunk 4]
```diff
@@ -30,18 +30,15 @@ define([
...
         button = t('button');
 
     function factory(config) {
-        let spec = config.parameterSpec,
+        let container, parent, ui;
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js:36]: separated let vs. const in a few modules to get eslint happier.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js [hunk 5]
```diff
@@ -207,18 +202,7 @@ define([
                     },
                 });
 
-                preButton = div(
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js:210]: This wasn't used anywhere -> deleted!

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/structView.js [hunk 6]
```diff
@@ -18,23 +18,18 @@ define([
...
+            structFields = {};
+        const spec = config.parameterSpec,
+            bus = config.bus,
             viewModel = {
                 data: {},
                 state: {
                     enabled: null,
                 },
             },
-            structFields = {},
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/view/structView.js:32]: also not actually used anywhere.

Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/appWidgets2/view/structView.js:32]: There's probably some more cleanup that could've been done here, too...

File: nbextensions/viewCell/widgets/viewCellWidget.js [hunk 7]
```diff
@@ -4,15 +4,13 @@ define(
         'bluebird',
         'uuid',
         'base/js/namespace',
-        'common/lang',
```
Comment (briehl) [inline nbextensions/viewCell/widgets/viewCellWidget.js:7]: not actually used here.

File: test/unit/spec/util/util-Spec.js [hunk 8]
```diff
@@ -73,5 +73,44 @@ define(['common/lang'], (Utils) => {
                 await expectAsync(Utils.pRequire([module])).toBeRejected();
             });
         });
+
+        describe('the toInteger function', () => {
```
Comment (briehl) [inline test/unit/spec/util/util-Spec.js:77]: added missing `toInteger` tests that should've been in the last PR. Oops!

**Review by ialarmedalien — APPROVED** [review]
🚀

**Pull request merged**